### PR TITLE
Add 'file' and 'rnm' to Synocli-File

### PR DESCRIPTION
**Pull request by hgy59**
- redesign cross/ncursesw to make independent of cross/ncurses
- create links for specific hashes of rhash
- add 'file' and 'rnm' to diyspk
- remove links on uninstall only if still linking to this package
- add creation of c++ library to cross/gmp

_Motivation:_  `rnm` is a rename tool without dependency of perl or python
_Linked issues:_  #3250, #2944

### Checklist
- [x] Build rule `all-supported` completed successfully
- [ ] Package upgrade completed successfully
- [x] New installation of package completed successfully

@adithyabenny can you verify whether rnm is worthful replacement for rename?
@fredrike can you confirm that creating links for special hashes is what it's ment for?

### CODE REVIEW

**Comment by ymartin59** [comment]
Would it be possible to include also "rmlint" #3482 and "detox" #2199 (one of them require scons but it can be added in Debian docker image)?

**Comment by digitalbox94** [comment]
Is "file" required for "rnm" ? Because "file" is already available in the synogear diagnostic tool (total of 127 commands available)

File: cross/rhash/PLIST [hunk 1]
```diff
@@ -1,12 +1,2 @@
-bin:bin/ed2k-link
```
Comment (ymartin59) [inline cross/rhash/PLIST]: Sure to remove other executables?

Comment (hgy59) [reply cross/rhash/PLIST]: yes, these are only links to bin/rhash and are created as links upon installation again.

File: cross/mc/PLIST [hunk 2]
```diff
@@ -2,7 +2,7 @@ bin:bin/mc
 bin:bin/mcdiff
 bin:bin/mcedit
 bin:bin/mcview
-rsc:etc/mc
+etc:etc/mc
```
[the remark this answers is not in the document]
Comment (hgy59) [reply cross/mc/PLIST]: sorry, my fault

**Comment by ymartin59** [comment]
I am including `rmlint` and `detox` too at the same time

**Pull request merged**